### PR TITLE
Add cmake 3.6.2 and clang/llvm7

### PR DESCRIPTION
**Pull request by jashook**

### CODE REVIEW

**Comment by jashook** [comment]
@omajid @jkoritzinsky ptal

**Comment by omajid** [comment]
I don't think this will work, unfortunately. The `scl ... bash` command starts a new bash session in which cmake is available on the $PATH. That session ends after the run command. I don't think the runtime build will find this cmake executible as it is, without further changes.

**Comment by jashook** [comment]
@omajid thank you for the review I have changed to use ENTRYPOINT in place of run. :)

**Comment by jashook** [comment]
ptal @janvorli

**Comment by jashook** [comment]
ping @omajid @janvorli @jkoritzinsky

File: src/centos/7/source-build/Dockerfile [hunk 1]
```diff
@@ -50,12 +50,7 @@ RUN yum install -y \
...
-# cmake will include 2.8
-
-RUN wget https://github.com/Kitware/CMake/releases/download/v3.6.2/cmake-3.6.2-Linux-x86_64.tar.gz && \
-    tar -xvf cmake-3.6.2-Linux-x86_64.tar.gz && \
-    mv cmake-3.6.2-Linux-x86_64 /cmake
-
 # Add to path
-ENV PATH "${PATH}:/cmake/bin"
\ No newline at end of file
+ENV PATH "${PATH}:/cmake/bin"
```
Comment (janvorli) [inline src/centos/7/source-build/Dockerfile:54]: It seems the /cmake/bin is not needed anymore.

Comment (janvorli) [reply src/centos/7/source-build/Dockerfile:54]: Otherwise it looks good.

**Comment by jashook** [comment]
@mthalman ptal

**Pull request merged**